dsgrid/registry/dimension_mapping_registry_manager.py

```diff
@@ -103,6 +107,31 @@ def check_unique_records(self, config: DimensionMappingsConfig, warn_only=False)
                     f"There are {len(duplicates)} dimension mappings with duplicate content (data files)."
                 )
 
+    def _check_records_against_dimension_records(self, config):
+        for mapping in config.model.mappings:
+            actual_from_records = {x.from_id for x in mapping.records}
+            actual_to_records = {x.to_id for x in mapping.records}
+            from_dimension = self._dimension_mgr.get_by_id(mapping.from_dimension.dimension_id)
+            allowed_from_records = from_dimension.get_unique_ids()
+            diff = actual_from_records.difference(allowed_from_records)
+            if diff:
+                dim_id = from_dimension.model.dimension_id
+                raise DSGInvalidDimensionMapping(
+                    f"Dimension mapping has invalid 'from' records: dimension_id={dim_id} {diff}"
+                )
+
+            # Note: this code cannot complete verify 'to' records. A dataset may be registering a
+            # mapping to a project's dimension for a specific data source, but that information
+            # is not available here.
+            to_dimension = self._dimension_mgr.get_by_id(mapping.to_dimension.dimension_id)
+            allowed_to_records = to_dimension.get_unique_ids()
+            diff = actual_to_records.difference(allowed_to_records)
+            if diff:
+                dim_id = from_dimension.model.dimension_id
+                raise DSGInvalidDimensionMapping(
+                    f"Dimension mapping has invalid 'to' records: dimension_id={dim_id} {diff}"
+                )
```

Comment:
I think the diff checking may need to be modified slightly to allow for null values? How are nulls going to show up in the actual_from_records and actual_to_records sets? Will they be excluded to begin with, show up as None, or as something else?

Reply:
Agreed. I just tested this for NULL values in the to-values and it fails.

Reply:
Fixed